### PR TITLE
Feature/handle coverage retention epmdj 10634

### PR DESCRIPTION
**Pull request by bodyangug**

### CODE REVIEW

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt [hunk 1]
```diff
@@ -27,16 +27,33 @@ import mu.KLogger
...
+abstract class RetentionCallbacks() : Endpoint(), MessageHandler.Whole<ByteArray> {
+    abstract fun setOnAvailable(callback: () -> Unit)
+    abstract fun setOnUnavailable(callback: () -> Unit)
+}
+
 class WsClientEndpoint(
     private val messageHandler: MessageHandler.Whole<Message>,
     private val reconnectHandler: WsClientReconnectHandler?
-) : Endpoint(), MessageHandler.Whole<ByteArray> {
+) : RetentionCallbacks() {
```
Comment (epamrsa) [inline agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt:39]: Why just not added third?

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt [hunk 2]
```diff
@@ -27,16 +27,33 @@ import mu.KLogger
 import mu.KotlinLogging
 import com.epam.drill.common.message.Message
 
+
+abstract class RetentionCallbacks() : Endpoint(), MessageHandler.Whole<ByteArray> {
```
Comment (epamrsa) [inline agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt:31]: Convert to interface, move to sepearate file

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt [hunk 3]
```diff
@@ -29,8 +29,8 @@ import com.epam.drill.common.message.MessageType
 
 object WsClient : WsClientReconnectHandler {
 
+    val endpoint = WsClientEndpoint(WsMessageHandler, this)
```
Comment (epamrsa) [inline agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt:32]: Why not private?

Comment (bodyangug) [reply agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt:32]: that field will be used by java-agent:
https://github.com/Drill4J/java-agent/pull/114/commits/5c40bf9ec4bb932f4b9eb80f38162e39bf06f547#diff-7df900a52df5b634c1ed5959c206a2665f975bc2a7f9819b8f8f756c20209b4eR49

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt (hunk 1 above)
Comment (bodyangug) [reply agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt]: done

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt (hunk 2 above)
Comment (bodyangug) [reply agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientEndpoint.kt]: done

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt (hunk 3 above)
Comment (bodyangug) [reply agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt]: fixed

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt [hunk 4]
```diff
@@ -27,12 +28,20 @@ import com.epam.drill.common.message.DrillMessageWrapper
 import com.epam.drill.common.message.Message
 import com.epam.drill.common.message.MessageType
 
-object WsClient : WsClientReconnectHandler {
+object WsClient : WsClientReconnectHandler, RetentionCallbacks {
```
Comment (RomanDavlyatshin) [inline agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt:31]: These callbacks have nothing to do with coverage retention task.
Rename to something like `ConnectionsStatusCallbacks`

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt (hunk 4 above)
Comment (bodyangug) [reply agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClient.kt]: done

**Pull request merged**